server/src/main/java/org/apache/iotdb/db/engine/tsfilemanagement/utils/HotCompactionUtils.java

```diff
@@ -180,44 +193,52 @@ public static void merge(RestorableTsFileIOWriter writer,
         for (Entry<String, MeasurementSchema> entry : deviceMeasurementEntry.getValue()
             .entrySet()) {
           String measurementId = entry.getKey();
-          Pair<ChunkMetadata, Chunk> chunkPair = writeSeqChunk(writer, storageGroup,
-              tsFileSequenceReaderMap, deviceId, measurementId, vmWriters, null, null);
+          Pair<ChunkMetadata, Chunk> chunkPair = writeSeqChunk(storageGroup,
+              tsFileSequenceReaderMap, deviceId, measurementId, tsFileResources, null,
+              null);
           ChunkMetadata newChunkMetadata = chunkPair.left;
           Chunk newChunk = chunkPair.right;
           if (newChunkMetadata != null && newChunk != null) {
             writer.writeChunk(newChunk, newChunkMetadata);
+            targetResource.updateStartTime(deviceId, newChunkMetadata.getStartTime());
+            targetResource.updateEndTime(deviceId, newChunkMetadata.getEndTime());
           }
         }
         writer.endChunkGroup();
-        if (vmLogger != null) {
-          vmLogger.logDevice(deviceId, writer.getPos());
+        if (hotCompactionLogger != null) {
```

Comment:
Could we keep the logger not null before calling this method?

Reply:
just confirm that the logger is not null